zerver/lib/test_classes.py

```diff
@@ -2052,11 +2052,11 @@ def capture_send_event_calls(
         # So explicitly change parameter name to 'notice' to work around this problem
         with (
             mock.patch("zerver.tornado.event_queue.process_notification", lst.append),
-            # Some `send_event` calls need to be executed only after the current transaction
-            # commits (using `on_commit` hooks). Because the transaction in Django tests never
-            # commits (rather, gets rolled back after the test completes), such events would
-            # never be sent in tests, and we would be unable to verify them. Hence, we use
-            # this helper to make sure the `send_event` calls actually run.
+            # Some `send_event_rollback_unsafe` calls need to be executed only after the
+            # current transaction commits (using `on_commit` hooks). Because the transaction
+            # in Django tests never commits (rather, gets rolled back after the test completes),
+            # such events would never be sent in tests, and we would be unable to verify them.
+            # Hence, we use this helper to make sure the `send_event_rollback_unsafe` calls actually run.
```

Comment:
I think actually this text is now confused; it's `send_event_on_commit` that requires this logic.

Reply:
I didn't understand what is confusing here ?

If we'll say:
```
Some `send_event_on_commit` calls need to be executed only after the current transaction commits...
```

This would be incorrect because `send_event_on_commit` gets executed the moment we call it and registers `on_commit` callback `send_event_rollback_unsafe`.

Reply:
```
Hence, we use this helper to make sure the `send_event_rollback_unsafe` calls actually run.
```

This line is also correct because `self.captureOnCommitCallbacks(execute=True),` makes sure that `send_event_rollback_unsafe` actually runs.

Reply:
Oh I see, I'll do a wording tweak and then merge this.